### PR TITLE
fix Debug component dependencies

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
         "symfony/class-loader": "~3.2",
         "symfony/dependency-injection": "^3.4.24|^4.2.5",
         "symfony/config": "~3.4|~4.0",
+        "symfony/debug": "~2.8|~3.0|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: `FrameworkBundle` registers `Symfony\Component\Debug\ErrorHandler` when `boot()` is called

File: src/Symfony/Bundle/TwigBundle/composer.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "require": {
         "php": "^5.5.9|>=7.0.8",
         "symfony/config": "~3.2|~4.0",
+        "symfony/debug": "~2.8|~3.0|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Bundle/TwigBundle/composer.json]: `ExceptionController::showAction()` has a parameter of type `Symfony\Component\Debug\Exception\FlattenException`

**Comment by xabbuh** [comment]
Before #31065, the Debug component was installed as a transitive dependency.

**Review by derrabus — APPROVED** [review]

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/Debug/Tests/FatalErrorHandler/ClassNotFoundFatalErrorHandlerTest.php [hunk 3]
```diff
@@ -61,7 +61,7 @@ public function testHandleClassNotFound($error, $translatedMessage, $autoloader
         }
 
         $this->assertInstanceOf('Symfony\Component\Debug\Exception\ClassNotFoundException', $exception);
-        $this->assertSame($translatedMessage, $exception->getMessage());
+        $this->assertRegExp($translatedMessage, $exception->getMessage());
```
Comment (stof) [inline src/Symfony/Component/Debug/Tests/FatalErrorHandler/ClassNotFoundFatalErrorHandlerTest.php]: why switching this to a regex, while all your regexps are anchored on both sides and contain only static parts ?

Comment (derrabus) [reply src/Symfony/Component/Debug/Tests/FatalErrorHandler/ClassNotFoundFatalErrorHandlerTest.php]: He added a `.*` in lines 103 and 121.

Comment (xabbuh) [reply src/Symfony/Component/Debug/Tests/FatalErrorHandler/ClassNotFoundFatalErrorHandlerTest.php]: error messages differ slightly for these two cases when the ErrorCatcher component is present: https://travis-ci.org/symfony/symfony/jobs/551399461#L4725

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**